### PR TITLE
Bug fix/remove js rebalance

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Remove JavaScript implementation of rebalancing.

- [ ] :hankey: Bugfix 
- [ ] :pizza: New feature 
- [x] :hammer: Refactoring 
- [ ] :book: CHANGELOG entry made
- [ ] :muscle: The behavior in this PR was *manually tested*
- [x] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [x] No backports required
- [ ] Backports required for: *(Please specify versions)*

### Testing & Verification

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [x] This change is already covered by existing tests, such as *resilience tests*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11932/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11934/

File: arangod/Sharding/ShardingStrategy.cpp [hunk 1]
```diff
@@ -36,6 +37,6 @@ bool ShardingStrategy::isCompatible(ShardingStrategy const* other) const {
 void ShardingStrategy::toVelocyPack(VPackBuilder& result) const {
   // only need to print sharding strategy if we are in a cluster
   if (ServerState::instance()->isRunningInCluster()) {
-    result.add("shardingStrategy", VPackValue(name()));
+    result.add(StaticStrings::ShardingStrategy, VPackValue(name()));
```
Comment (maierlars) [inline arangod/Sharding/ShardingStrategy.cpp:40]: this could have been a delete-only pr :disappointed:

**Review by maierlars — APPROVED** [review]
-

File: arangod/Sharding/ShardingStrategy.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Sharding/ShardingStrategy.cpp:40]: @maierlars : still a net deletion of 177 LOC :+1:

**Comment by curtdept** [comment]
@jsteemann whats replacing it?

**Comment by jsteemann** [comment]
@curtdept : the functionality was moved from a JavaScript to a C++ implementation in a previous commit (3d7e2ba3b97b666584b646fffd85f1705ae5c136).

**Comment by curtdept** [comment]
@jsteemann oh nice thanks :)

**Pull request merged**